Problem2.java

```diff
@@ -0,0 +1,36 @@
+// Time Complexity : Need help
```

Comment:
O(n) is the time complexity for your solution
as you are iterating over the preorder array once!